openpype/plugins/publish/integrate_new.py

```diff
@@ -113,816 +159,538 @@ class IntegrateAssetNew(pyblish.api.InstancePlugin):
+            "type": "version",
+            "parent": subset["_id"],
+            "name": version_number,
+            "data": self.create_version_data(instance)
+        }
 
-            Uses Settings > Global > Publish plugins > IntegrateAssetNew
+        existing_version = io.find_one({
+            'type': 'version',
+            'parent': subset["_id"],
+            'name': version_number
+        }, projection={"_id": True})
 
-            Args:
-                instance (dict): processed instance
-                subset_id (str): DB's subset _id
+        if existing_version:
+            self.log.debug("Updating existing version ...")
+            version["_id"] = existing_version["_id"]
+        else:
+            self.log.debug("Creating new version ...")
+            version["_id"] = ObjectId()
+
+        bulk_writes = [ReplaceOne(
+            filter={"_id": version["_id"]},
+            replacement=version,
+            upsert=True
+        )]
+
+        self.log.info("Prepared version: v{0:03d}".format(version["name"]))
+
+        return version, bulk_writes
+
+    def prepare_representation(self, repre,
+                               template_name,
+                               existing_repres_by_name,
+                               version,
+                               instance_stagingdir,
+                               instance):
+
+        # pre-flight validations
+        if repre["ext"].startswith("."):
+            raise ValueError("Extension must not start with a dot '.': "
+                             "{}".format(repre["ext"]))
+
+        if repre.get("transfers"):
+            raise ValueError("Representation is not allowed to have transfers"
+                             "data before integration. They are computed in "
+                             "the integrator"
+                             "Got: {}".format(repre["transfers"]))
+
+        # create template data for Anatomy
+        template_data = copy.deepcopy(instance.data["anatomyData"])
+
+        # required representation keys
+        files = repre['files']
+        template_data["representation"] = repre["name"]
+        template_data["ext"] = repre["ext"]
+
+        # optionals
+        # retrieve additional anatomy data from representation if exists
+        for key, anatomy_key in {
+            # Representation Key: Anatomy data key
+            "resolutionWidth": "resolution_width",
+            "resolutionHeight": "resolution_height",
+            "fps": "fps",
+            "outputName": "output",
+            "originalBasename": "originalBasename"
+        }.items():
+            # Allow to take value from representation
+            # if not found also consider instance.data
+            if key in repre:
+                value = repre[key]
+            elif key in instance.data:
+                value = instance.data[key]
+            else:
+                continue
+            template_data[anatomy_key] = value
```

Comment:
After a recent merge conflict with latest develop which introduced `originalBasename` I refactored this code to first consider representation, if not found then consider `instance.data` values (which is where `originalBasename` was taking it from in latest develop).

This change should also allow to pick up `fps` from `instance.data` too which I believe was also in the old integrator code.

Reply:
This does change the logic that a null value or zero value can still be set in `template_data`. Previously values were check with `if value:` - now they aren't. Thoughts on this?